app/services/sfn_pipeline_dispatch_service.rb

```diff
@@ -124,43 +109,56 @@ def convert_dag_json_to_wdl(dag_json)
+      POSTPROCESS_WDL_URI: "s3://idseq-workflows/v#{@sfn_tags[:wdl_version]}/main/postprocess.wdl",
+      EXPERIMENTAL_WDL_URI: "s3://idseq-workflows/v#{@sfn_tags[:wdl_version]}/main/experimental.wdl",
       Input: {
-        HostFilter: input_files_paths.each_with_index.map { |path, i| ["fastqs_#{i}", path] }.to_h,
+        HostFilter: {
+          fastqs_0: File.join(@sample.sample_input_s3_path, @sample.input_files[0].name),
+          fastqs_1: @sample.input_files[1] ? File.join(@sample.sample_input_s3_path, @sample.input_files[1].name) : nil,
+          file_ext: @sample.fasta_input? ? 'fasta' : 'fastq',
+          nucleotide_type: @sample.metadata.find_by(key: "nucleotide_type").string_validated_value,
+          host_genome: @sample.host_genome_name.downcase,
+          adapter_fasta: PipelineRun::ADAPTER_SEQUENCES[@sample.input_files[1] ? "paired-end" : "single-end"],
+          star_genome: @sample.s3_star_index_path,
+          bowtie2_genome: @sample.s3_bowtie2_index_path,
+          human_star_genome: HostGenome.find_by(name: "Human").s3_star_index_path,
+          human_bowtie2_genome: HostGenome.find_by(name: "Human").s3_bowtie2_index_path,
+          max_input_fragments: @pipeline_run.max_input_fragments,
+          max_subsample_fragments: @pipeline_run.subsample,
+        }, NonHostAlignment: {
+          lineage_db: @pipeline_run.alignment_config.s3_lineage_path,
+          accession2taxid_db: @pipeline_run.alignment_config.s3_accession2taxid_path,
+          taxon_blacklist: @pipeline_run.alignment_config.s3_taxon_blacklist_path,
+          index_dir_suffix: @pipeline_run.alignment_config.index_dir_suffix,
+          deuterostome_db: @sample.skip_deutero_filter_flag ? nil : @pipeline_run.alignment_config.s3_deuterostome_db_path,
+          use_taxon_whitelist: @pipeline_run.use_taxon_whitelist,
+        }, Postprocess: {
+          nt_db: @pipeline_run.alignment_config.s3_nt_db_path,
+          nt_loc_db: @pipeline_run.alignment_config.s3_nt_loc_db_path,
+          nr_db: @pipeline_run.alignment_config.s3_nr_db_path,
+          nr_loc_db: @pipeline_run.alignment_config.s3_nr_loc_db_path,
+          lineage_db: @pipeline_run.alignment_config.s3_lineage_path,
+          taxon_blacklist: @pipeline_run.alignment_config.s3_taxon_blacklist_path,
+          deuterostome_db: @sample.skip_deutero_filter_flag ? nil : @pipeline_run.alignment_config.s3_deuterostome_db_path,
+          use_taxon_whitelist: @pipeline_run.use_taxon_whitelist,
+        }, Experimental: {
+          nt_db: @pipeline_run.alignment_config.s3_nt_db_path,
+          nt_loc_db: @pipeline_run.alignment_config.s3_nt_loc_db_path,
+          file_ext: @sample.fasta_input? ? 'fasta' : 'fastq',
+          nt_info_db: @pipeline_run.alignment_config.s3_nt_info_db_path || PipelineRunStage::DEFAULT_S3_NT_INFO_DB_PATH,
+          use_taxon_whitelist: @pipeline_run.use_taxon_whitelist,
+        },
```

Comment:
Could you store these input variables in the `PipelineRun` or write them in a file to S3 so I could access them for the PipelineViz?

Reply:
The inputs are stored in the step function state and in S3 at
```
s3://idseq-samples-${deployment_environment}/samples/${project_id}/${sample_id}/results/idseq-dev-main-1/wdl-${wdl_version}/dag-${dag_version}/host_filter_input.json
s3://idseq-samples-${deployment_environment}/samples/${project_id}/${sample_id}/results/idseq-dev-main-1/wdl-${wdl_version}/dag-${dag_version}/non_host_alignment_input.json
s3://idseq-samples-${deployment_environment}/samples/${project_id}/${sample_id}/results/idseq-dev-main-1/wdl-${wdl_version}/dag-${dag_version}/postprocess_input.json
s3://idseq-samples-${deployment_environment}/samples/${project_id}/${sample_id}/results/idseq-dev-main-1/wdl-${wdl_version}/dag-${dag_version}/experimental_input.json
```